[SPARK-23759][UI] Unable to bind Spark2 history server to specific host name / IP

felixalbani opened the pull request:
## What changes were proposed in this pull request?

This pull is to fix SPARK-23759 issue

Problem was created due connector.setHost(hostName) call was after connector.start()

## How was this patch tested?

Patch was tested after build and deployment. This patch requires SPARK_LOCAL_IP environment variable to be set on spark-env.sh

Please review http://spark.apache.org/contributing.html before opening a pull request.

Commit:
Solution to SPARK-23759 is to setHost before starting the connector

Solution to SPARK-23759 is to setHost before starting the connector.

I run few tests and was able to confirm the binding happens as expected.

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Can one of the admins verify this patch?

Commit:
Revert "Solution to SPARK-23759 is to setHost before starting the connector"

This reverts commit f9d1293991b34a80b4fa11a846812e3a79e1493f.

wangyum commented:
Please update the title.

Commit:
[SPARK-23759] [UI] Unable to bind Spark2 history server to specific host name / IP

Resolves SPARK-23759 simply by moving call to setHost function before connector start

mgaido91 commented on core/src/main/scala/org/apache/spark/ui/JettyUtils.scala:
```diff
@@ -330,12 +330,13 @@ private[spark] object JettyUtils extends Logging {
 
         // Currently we only use "SelectChannelConnector"
         // Limit the max acceptor number to 8 so that we don't waste a lot of threads
         connector.setAcceptQueueSize(math.min(connector.getAcceptors, 8))
```
I think also this line should be moved before the start.

mgaido91 commented:
cc @vanzin 

I think this comes from SPARK-17874. I see that some properties (the connector name in particular) are set after this method. So I guess they have no effect too. Shall we move them here before the start too?

vanzin commented:
Yes, that's a bug (and also is being fixed as part of #20327, but ok if you want to fix it here).

Commit:
[SPARK-23759][UI] Unable to bind Spark2 history server to specific host name / IP

Resolves SPARK-23759 simply by moving call to setHost function before connector start

felixalbani commented:
@mgaido91 I pushed the start() after setAcceptQueueSize in new commit

vanzin commented:
ok to test

vanzin commented:
Actually, this PR is against 2.2, not master. Please close this and open a new one against the right branch.

SparkQA commented:
**[Test build #88527 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/88527/testReport)** for PR 20867 at commit [`5d7ed7a`](https://github.com/apache/spark/commit/5d7ed7ae933d2dc473abdb6f0fc0277d5cf0d7da).

vanzin commented:
I don't know, it didn't use to be possible to change the base branch. If it is now, just change it to master.
